crates/proto/src/rr/record_data.rs

```diff
@@ -691,6 +691,9 @@ pub enum RData {
         rdata: NULL,
```

Comment:
one extra comment about the `NULL` type: I think it should be called something else (more descriptive) since it's essentially a newtype over raw RData bytes (`Vec<u8>`) that have not been deserialized / decoded.

I find the "NULL" name confusing because there's also a NULL record type (rtype = 10) described in RFC1035. this `RData::Unknown` is not about that NULL record, however, because the NULL record is covered by the variant `RData::NULL`.

IOW, I think this would read better as

``` rust
pub enum RData {
    NULL(Opaque),
    Unknown {
        code: RecordType,
        rdata: Opaque,
    }
    // ..
}

/// RData payload that has not been deserialized / decoded
pub struct Opaque { // or "Bytes" or "RDataPayload"
    inner: Vec<u8>
}
```
but that's a refactor / rename for later